### ISSUE TITLE
Wordcount example broken

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#215]
I guess introduced by https://github.com/stratosphere/stratosphere/pull/195

Calling Wordcount without any arguments causes a NPE.

Calling it with the correct arguments (`2  hdfs:///user/robert/datasets/hamlet-wc hdfs:///tmp/stratosphere-dev/wc`)

Causes this 

```
Could not find or load main class 2
```

No stacktrace.

**Comment by rmetzger** [comment]
I think we should set the LocalExecutor as the default execution method for WordCount.

The current variant requires the user to setup stratosphere and start it in Local mode.

**Comment by rmetzger** [comment]
The `LocalExecutor` does not work as well:

```
Plan plan = wc.getPlan("1", "hdfs:///user/robert/datasets/hamlet-wc", "hdfs:///tmp/stratosphere-dev/wc");
LocalExecutor.execute(plan);
```

Executing the wordcount example does not work on the cluster:

```
/stratosphere/bin/pact-client.sh run\
 --jarfile ./stratosphere/examples/pact/pact-examples-0.4-SNAPSHOT-WordCount.jar \
 --arguments $DOP hdfs:///user/robert/datasets/hamlet-wc hdfs:///tmp/stratosphere-dev/wc
```

results in

```
ERROR: The job was not successfully submitted to the nephele job manager: eu.stratosphere.nephele.executiongraph.GraphConversionException: Cannot compute input splits for Input Lines: java.io.IOException: Incomplete HDFS URI, no host: file:///
    at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:96)
    at eu.stratosphere.nephele.fs.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:131)
    at eu.stratosphere.nephele.fs.FileSystem.get(FileSystem.java:224)
    at eu.stratosphere.nephele.fs.Path.getFileSystem(Path.java:298)
    at eu.stratosphere.pact.generic.io.FileInputFormat.createInputSplits(FileInputFormat.java:322)
    at eu.stratosphere.pact.generic.io.FileInputFormat.createInputSplits(FileInputFormat.java:71)
    at eu.stratosphere.pact.runtime.task.DataSourceTask.computeInputSplits(DataSourceTask.java:326)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.createVertex(ExecutionGraph.java:554)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.constructExecutionGraph(ExecutionGraph.java:275)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.<init>(ExecutionGraph.java:176)
    at eu.stratosphere.nephele.jobmanager.JobManager.submitJob(JobManager.java:511)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at eu.stratosphere.nephele.ipc.RPC$Server.call(RPC.java:417)
    at eu.stratosphere.nephele.ipc.Server$Handler.run(Server.java:946)

    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.createVertex(ExecutionGraph.java:557)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.constructExecutionGraph(ExecutionGraph.java:275)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.<init>(ExecutionGraph.java:176)
    at eu.stratosphere.nephele.jobmanager.JobManager.submitJob(JobManager.java:511)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at eu.stratosphere.nephele.ipc.RPC$Server.call(RPC.java:417)
    at eu.stratosphere.nephele.ipc.Server$Handler.run(Server.java:946)
```

I think I've done nothing wrong.

**Comment by StephanEwen** [comment]
Part one fixed in b2af1d9a1bd3d1af6ae166925c069d5febdf0eba

**Comment by StephanEwen** [comment]
The problem that the hdfs path do not work is the following:
You need either a compete hdfs uri that specifies host and port of the namenode. Or you need a default hdfs configuration registered with stratosphere, in the config file.

We need a much better error message, though!

**Comment by StephanEwen** [comment]
The local executor cannot run with HDFS, because the distributed file system is in the nephele-hdfs project, which is not in the dependency path. I will add it there...

**Comment by rmetzger** [comment]
Just as a documentation:
You need to set 
`fs.hdfs.hdfsdefault: /share/hadoop/hadoop/conf/hdfs-site.xml` to your Stratosphere config.

And this file must contain the value

```
<property><name>fs.default.name</name><value>hdfs://namenodeHost:40010/</value></property>
```

Or you specify the namenodeHost with your file URI:
`hdfs://namenodeHost:40010/fancyFile.seq`

**Comment by aljoscha** [comment]
@StephanEwen fixed this, right?

**Comment by StephanEwen** [comment]
Should be fixed in 159088dc0abf0e217a0441badea0e899a488b31b and b2af1d9a1bd3d1af6ae166925c069d5febdf0eba
